### PR TITLE
209 Push to stream

### LINKED ISSUE #209: Add PushToStreamExpr
Pushing value to a stream is a very common pattern, used for conditional returns and aggregation.

Currently, the only way to push a value to a stream is via a function call. But functions have strict types, so to keep the type checker happy we need to make a new function with proper types every time. It makes the code very cumbersome.

In the future, we should use (co)functors and `(call` profunctor optics to fix that better way: with no function execution, keeping the tetraplets untouched. Yet to do that, this pattern should be well identified in the code.

The proposition is to add custom syntax for pushing a value into a stream:

```
-- It must be a stream, defined in the scope
stream: *u32
-- Push a value into the stream: value may be a literal, a variable, constant, whatever
stream <<- value
```

Now it can be compiled with `Op.identity`. Later we will optimize with `(apply` or some other advanced AIR syntax.

Related to #200, #187

### PR DESCRIPTION
**Pull request by DieMyst**
fixes #209

### CODE REVIEW

File: parser/src/main/scala/aqua/parser/expr/PushToStreamExpr.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import aqua.parser.Expr
+import aqua.parser.lexer.Token._
+import aqua.parser.lexer.{Value, VarLambda}
+import aqua.parser.lift.LiftParser
+import cats.Comonad
+import cats.parse.{Parser => P}
+
+case class PushToStreamExpr[F[_]](
+  stream: VarLambda[F],
```
Comment (alari) [inline parser/src/main/scala/aqua/parser/expr/PushToStreamExpr.scala:11]: You just need to read `Name[F]`, as you can't write to a stream with lambda. E.g. `stream! <<- "val"` is incorrect.

File: semantics/src/main/scala/aqua/semantics/expr/PushToStreamSem.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  def program[Alg[_]](implicit
+    N: NamesAlgebra[F, Alg],
+    T: TypesAlgebra[F, Alg],
+    V: ValuesAlgebra[F, Alg]
+  ): Prog[Alg, Model] =
+    V.valueToModel(expr.value).flatMap {
+      case Some(vm) =>
+        for {
+          resolvedStreamType <- N.read(expr.stream.name).map(_.get)
```
Comment (alari) [inline semantics/src/main/scala/aqua/semantics/expr/PushToStreamSem.scala:23]: This is unsafe.

File: semantics/src/main/scala/aqua/semantics/expr/PushToStreamSem.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          valueType <- V.resolveType(expr.value)
+          _ <- T.expectStream(expr.stream, resolvedStreamType)
+          _ <- T.expectStreamMember(expr.value, resolvedStreamType, valueType)
```
Comment (alari) [inline semantics/src/main/scala/aqua/semantics/expr/PushToStreamSem.scala:26]: You can build both checks from `ValuesAlgebra.ensureTypeMatches`

File: semantics/src/main/scala/aqua/semantics/expr/PushToStreamSem.scala [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        } yield {
+          FuncOp
+            .leaf(
+              CallServiceTag(
```
Comment (alari) [inline semantics/src/main/scala/aqua/semantics/expr/PushToStreamSem.scala:30]: Use `FuncOps.identity` builder instead.

**Review by alari — CHANGES_REQUESTED** [review]

**Review by alari — APPROVED** [review]

**Comment by DieMyst** [comment]
fixes #209

**Pull request merged**